src/pages/api/register.ts

```
@@ -0,0 +1,97 @@
+    } else if (confirmPassword !== password) {
+        passwordErrorMessage = 'Passwords do not match';
+    }
+
+    return passwordErrorMessage;
+};
+
+export default async (req: NextApiRequest, res: NextApiResponse): Promise<void> => {
+    const setErrorsCookie = (inputChecks: InputCheck[], regKey: string) => {
+        const cookieContent = JSON.stringify({ inputChecks });
+        setCookieOnResponseObject(getDomain(req), USER_COOKIE, cookieContent, req, res);
+        redirectTo(res, `/register?key=${regKey}`);
+    };
+
+    try {
+        const { email, password, confirmPassword, nocCode, regKey } = req.body;
+
+        const inputChecks: InputCheck[] = [];
+
+        const emailValid = checkEmailValid(email);
+
+        inputChecks.push({
+            inputValue: !emailValid ? '' : email,
+            id: 'email',
+            error: !emailValid ? 'Enter an email address in the correct format, like name@example.com' : '',
+        });
+
+        inputChecks.push({
+            inputValue: '',
+            id: 'password',
+            error: validatePassword(password, confirmPassword),
+        });
+
+        inputChecks.push({
+            inputValue: nocCode,
+            id: 'nocCode',
+            error: nocCode === '' ? 'NOC cannot be empty' : '',
+        });
+
+        if (nocCode !== '') {
+            const servicesForNoc = await getServicesByNocCode(nocCode);
+
+            if (servicesForNoc.length === 0) {
+                inputChecks.push({
+                    inputValue: '',
+                    id: 'email',
+                    error: 'There was a problem logging you in',
+                });
+            }
+        }
+
+        if (inputChecks.some(el => el.error !== '')) {
+            setErrorsCookie(inputChecks, regKey);
+            return;
+        }
+
+        try {
+            const user = await Auth.signIn(email, regKey);
+
+            if (user.challengeName === 'NEW_PASSWORD_REQUIRED') {
```

We could probably do with an else on this to throw an error: `throw new Error('unexpected challenge: ', user.challengeName)`